site/content/contribute/webapp/migrating-to-typescript.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Migrating to Typescript"
+date: 2020-07-11T23:00:00-04:00
+weight: 9
+subsection: Web App
+---
+
+The Mattermost team wants to improve the quality, security and stability of the code, and one way to do this is by introducing the usage of type checking. Thus, we have decided to introduce Typescript in our codebase. As a first step, we have migrated the [mattermost-redux](https://github.com/mattermost/mattermost-redux) library to use Typescript, and are now in the processs of migrating the [webapp](https://github.com/mattermost/mattermost-webapp) to use Typescript.
+
+This campaign will help with the migration by converting files written in Javascript to type-safe files written in Typescript.
+
+By completing this campaign, we are looking to:
+
+- reduce the errors derived from changes.
+- increase the consistency of the code.
+- ensure a more defensive programming in the code.
+
+## Contributing
+
+If you're interested in contributing, please join the [Typescript Migration channel on community.mattermost.com](https://community.mattermost.com/core/channels/typescript-migration). You can also check out the [Contributors](https://community.mattermost.com/core/channels/tickets) channel, where there are several posts mentioning tickets related to this campaign, each containing the hashtag `#typescriptmigration`. You can work on migrating an individual module to Typescript by claiming a ticket that matches [this GitHub issue search](https://github.com/mattermost/mattermost-server/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Area%2FTechnical+Debt%22+label%3A%22Up+For+Grabs%22+Migrate+to+Typescript).
+
+For guidance on migrating a file to Typescript, please read the next section.
+
+## Component Migration Steps
+
+There are a few steps involved with migrating a file to use Typescript. Some of them may not apply to every file and they may change slightly based on the file you're working on. In general, you can follow these steps as a checklist for work that needs to be done on each file.
+
+1. Every React component's set of `props` need to be converted to a new type. You can use the component's `propTypes` as a template for what `props` a given component can expect. This conversion to Typescript includes maintaining whether a given `prop` is required or not. An optional property is Typescript is noted by including a question mark at the end of the property's name.
```

Comment:
```suggestion
1. Every React component's set of `props` needs to be converted to a new type. You can use the component's `propTypes` as a template for what `props` a given component can expect. This conversion to Typescript includes maintaining whether a given `prop` is required or not. An optional property is Typescript is noted by including a question mark at the end of the property's name.
```

Reply:
Confirm whether last sentence is should read similar to: 'An optional property is `Typescript` and is noted by including a question mark at the end of the property's name.' > unclear which property's name needs to have the question mark appended.

Reply:
I have a typo there. It should say "An optional property _in_ Typescript". Here's the sentence with some emphasis so the sentence is more clear to you for review:

An _optional property_ in Typescript is noted by including a question mark at the end of the _property's_ name.

Reply:
Thanks - makes perfect sense. :)